src/lib/utils/options/declaration.ts

```diff
@@ -199,6 +211,12 @@ export interface BooleanDeclarationOption extends DeclarationOptionBase {
      * If not specified defaults to false.
      */
     defaultValue?: boolean;
+
+    /**
+     * An optional validation function that validates a potential value of this option.
+     * The function must throw an Error if the validation fails and should do nothing otherwise.
+     */
+    validate?: (value: boolean) => void;
```

Comment:
I don't think we want a validate function here. It doesn't make sense to me to validate a true/false value.... If only one is allowed, it should not be an option.

Reply:
I was thinking about this too. I figured that one might want to validate based on some other value(s), eg: the value of another option, the current date :wink: or whatever. What do you think?

Reply:
Hmm... at that point, I think plugin authors would be better off doing validation when they are going to use the options. There's no guarantee about the order which options are validated in, so doing this would likely result in duplicating code.

Reply:
Okay, you have the last word.